lib/gen_stage.ex

```diff
@@ -318,12 +334,13 @@ defmodule GenStage do
 
   In many situations, producers may attempt to emit events while no consumers
   have yet subscribed. Similarly, consumers may ask producers for events
-  that are not yet available. In such cases, it is necessary for producers
-  to buffer events until a consumer is available or buffer the consumer
-  demand until events arrive, respectively. As we will see next, buffering
-  events can be done automatically by `GenStage`, while buffering the demand
-  is a case that must be explicitly considered by developers implementing
-  producers.
+  that are not yet available. Lastly, producers may emit more events than the
+  consumer sends demand for. In the first two cases, it is necessary for producers
+  to buffer events until a consumer is available or has demand. In the last case,
```

Comment:
I think the order here is wrong.

> producers may attempt to emit events while no consumers have yet subscribed

This means producer buffers events.

> Similarly, consumers may ask producers for events that are not yet available

This means producer buffers demand.

> Lastly, producers may emit more events than the consumer sends demand for.

This means producer buffers events.

---

Maybe we should break into two paragraphs, each dealing with one particular "excess" and explaining when it happens and what we can do about it?